Rd20266 update response webhook payload abo (#75)
ringcentral/engage-digital-api-docs: docs/webhooks/events.md

!!! warning "New content events"
    To be notified of new content from customers you should watch for content.imported but be careful as you will receive more than just customer contents.

>     In particular the ​`content.imported` event means imported from an Engage Digital point of view, it notifies all interactions coming directly from the synchronized source. For example you publish a post on a managed facebook page without using Engage Digital, a content.imported webhook will be triggered. For the Chat automatic welcome message (initial message) of the conversation will also be imported. So it’s not only customer message, this is a bit more complicated and some filtering may be required. To ignore those messages you will filter based on the status field to segregate actionable contents from customer Vs source itself.

Author: Done in 726673c

Property | Value | Description
---------|-------|------------
`answers` | Hash | Survey question and responses
`submitted_at` | date-time | Date/time response was submitted (EST/EDT or GMT -5/GMT -4)
`response_id` | String | Survey external Response ID
`url` | Hash | URL variable value for Survey
`main_indicator` | Integer | Main question result
`main_indicator_scaled` | Float | Main question result scaled to 0..1
`intervention_id` | String | ID of the intervention related to this survey
`survey_id` | String | ID of the survey
`source_id` | String | ID of the source related to this survey
`user_id` | String | ID of the user related to this survey

> `context_data` | Hash | Context_data is present only if the content has context_data associated. The context_data hash keys are the custom fields keys.

Author: Never too late to update it 🙃

> `questions` | Array of Hashes | List of questions and their responses. Each entry contains two keys: `text` and `value`

Reviewer: > Each entry contains two keys: `text` and `value`

this part is not exactly right, here's the structure of the `questions` field:
```json
[
  {
    "foreign_id": "free_text_question",
    "text": "What can we do better next time?",
    "replies": [
      {
        "text": "Answer faster",
        "value": "Answer faster"
      }
    ]
  },
  {
    "foreign_id": "multiple_choice_question",
    "text": "What do you like to eat?",
    "replies": [
      {
        "text": "Fish",
        "value": "3"
      },
      {
        "text": "Eggs",
        "value": "4"
      }
    ]
  },
  {
    "foreign_id": "single_choice_question",
    "text": "How would you rate our response?",
    "replies": [
      {
        "text": "Good",
        "value": "5"
      }
    ]
  }
]
```

IMO we should specify that each element of the array represents a question and inside this element there's another array of hashes that represents each of the replies to the given question

Author: Oh yeah it's not very clear, let me fix that

Author: Did it in ebb77e9), I also updated the screenshot above of the description of the PR if you want to see the final result.

> `questions` | Array of Hashes | List of questions and their responses. Each entry contains two keys : `text` and `value`

Reviewer: ```suggestion
`questions` | Array of Hashes | List of questions and their responses. Each entry contains two keys: `text` and `value`
```

Author: Done in 726673c

> `response_foreign_id` | String | `[Deprecated]`. Will be renamed `foreign_id`. Internal Response ID given by the provider

Author: Oh yeah somehow I thought we were renaming it to `foreign_id` in https://jira.ringcentral.com/browse/RD-21262, but that's just not true.

> `questions` | Array of Hashes | List of questions containing the displayed `text` of the question, its `foreign_id` (from Alchemer) and their `replies` in the form of an Array. Each one of the `replies` entry contains two keys: `text` and `value` of the customer reply.

Reviewer: this webhook is not only about Alchemer so we should be more generic:
```suggestion
`questions` | Array of Hashes | List of questions containing the displayed `text` of the question, its `foreign_id` (from the third party survey supplier) and their `replies` in the form of an Array. Each one of the `replies` entry contains two keys: `text` and `value` of the customer reply.
```

Author: 🤦  thanks

> `foreign_id` | String | `[Deprecated]`. Internal Response ID generated by Engage Digital.

Reviewer: ```suggestion
`foreign_id` | String | `[Deprecated]` Please use `id` instead. Internal Response ID generated by Engage Digital.
```

Reviewer: BTW you can probably do the same for `answers` -> `questions`

Author: Good call, done along with answers part in 726673c

> `response_foreign_id` | String | Internal Response ID fetched from the provider.

Reviewer: ```suggestion
`response_foreign_id` | String | ID of the response on the third party service.
```
